### PR TITLE
MBM-316 Electric defensive all tiers and Electric Debuff

### PR DESCRIPTION
**Pull request by MattKirchh**

### CODE REVIEW

File: src/main/java/com/mineboundteam/minebound/effect/registry/EffectRegistry.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import net.minecraftforge.registries.RegistryObject;
+
+import java.awt.*;
+
+public class EffectRegistry {
+    public static final DeferredRegister<MobEffect> MOB_EFFECTS = DeferredRegister.create(ForgeRegistries.MOB_EFFECTS,
+            MineBound.MOD_ID);
+
+    public static final RegistryObject<MobEffect> ELECTRIC_DEBUFF = MOB_EFFECTS.register("electric_debuff",
+            () -> new ElectricDebuff(MobEffectCategory.HARMFUL, new Color(234, 232, 166).getRGB()));
```
Comment (insertStringHere) [inline src/main/java/com/mineboundteam/minebound/effect/registry/EffectRegistry.java:18]: Should this color be in the Tooltip Util?

File: src/main/java/com/mineboundteam/minebound/magic/DefensiveSpells/ElectricDefensiveSpell.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+                entity.addEffect(new MobEffectInstance(EffectRegistry.ELECTRIC_DEBUFF.get(), spell.durationInTicks, spell.level.getValue()));
+                reduceMana(spell.manaCostPerHit, player);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public void appendHoverText(ItemStack pStack, @Nullable Level pLevel, List<Component> pTooltipComponents, TooltipFlag pIsAdvanced) {
```
Comment (insertStringHere) [inline src/main/java/com/mineboundteam/minebound/magic/DefensiveSpells/ElectricDefensiveSpell.java:91]: Should wait for Tooltip changes, or tooltip changes should wait for this

File: src/main/java/com/mineboundteam/minebound/magic/DefensiveSpells/ElectricDefensiveSpell.java [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+                spellTriggered = triggerSpell(player, entity, getSelectedSpell(player, PlayerSelectedSpellsProvider.SECONDARY_SPELL));
+            }
+            if (!spellTriggered) {
+                triggerSpell(player, entity, player.getUseItem());
+            }
+        }
+    }
+
+    protected static boolean triggerSpell(Player player, LivingEntity entity, ItemStack selectedSpell) {
+        if (selectedSpell.getItem() instanceof ElectricDefensiveSpell spell) {
```
Comment (insertStringHere) [inline src/main/java/com/mineboundteam/minebound/magic/DefensiveSpells/ElectricDefensiveSpell.java:80]: Might wanna ensure entity != player

**Review by insertStringHere — APPROVED** [review]

File: src/main/java/com/mineboundteam/minebound/magic/DefensiveSpells/ElectricDefensiveSpell.java (hunk 2 above)
Comment (MattKirchh) [reply src/main/java/com/mineboundteam/minebound/magic/DefensiveSpells/ElectricDefensiveSpell.java:91]: #174 will probably wait for this and #173.

File: src/main/java/com/mineboundteam/minebound/effect/registry/EffectRegistry.java (hunk 1 above)
Comment (MattKirchh) [reply src/main/java/com/mineboundteam/minebound/effect/registry/EffectRegistry.java:18]: I can add it to the `ColorUtil` once this is merged.

**Review by YonatanWN — APPROVED** [review]

**Pull request merged**